gdal/doc/source/download.rst

```diff
@@ -59,8 +59,12 @@ Windows builds are available via `Conda Forge`_ (64-bit only). See the
 
 .. _`Conda Forge`: https://anaconda.org/conda-forge/gdal
 
+Alternatively, [Tamas Szekeres](https://github.com/szekerest) provides packages
+with daily builds of latest stable releases as well as maintenance and
+development branches on his `GISInternals`_ site.
+
+.. _`GISInternals`: https://www.gisinternals.com
 
```

Comment:
If we're adding this, we should add MS4W and any others that want to be added too. See #1641 for discussion.

Reply:
I actually wondered why none of those are listed in the new docs and I decided to provoke some answers with this PR (I failed to find the other one for MS4W). 

In #1641 you say:
> I think we only want to list builds that have fully reproducible open source build recipes

It makes sense. I'm closing this PR and leaving it to @szekerest to re-open, if he feels determined to have his packages advertised.